kolibri/core/tasks/queue.py

```diff
@@ -66,6 +66,23 @@ def update_progress(progress, total_progress, stage=""):
         job_id = self.storage.enqueue_job(job, self.name)
         return job_id
 
+    def restart_job(self, job_id):
+        """
+        Given a job_id, restart the job for that id. A job will only be restarted if
+        in CANCELED or FAILED state.
+
+        This will create a new job with a new job_id.
+        """
+
+        old_job = self.fetch_job(job_id)
+        print(old_job.state)
+        if old_job.state in [State.CANCELED, State.FAILED]:
+            self.clear_job(job_id)
+            job = Job(old_job)
+            return self.enqueue(job)
+        else:
+            return None
```

Comment:
Feels like we should raise some sort of error here if someone is trying to restart a job that was not cancelled or did not fail?

We can then handle that error in the API endpoint and return a `ValidationError`.

Reply:
Yup. Wasn't sure about this. Added this with a new exception `JobNotRestartable`